lib/fog/google/models/storage_json/file.rb

```diff
@@ -0,0 +1,154 @@
+require "fog/core/model"
+
+module Fog
+  module Storage
+    class GoogleJSON
+      class File < Fog::Model
+        identity :key, :aliases => "Key"
+
+        # TODO: Verify
+        attribute :acl
+        attribute :predefined_acl
+        attribute :cache_control,       :aliases => "cacheControl"
+        attribute :content_disposition, :aliases => "contentDisposition"
+        attribute :content_encoding,    :aliases => "contentEncoding"
+        attribute :content_length,      :aliases => "size", :type => :integer
+        attribute :content_md5,         :aliases => "md5Hash"
+        attribute :content_type,        :aliases => "contentType"
+        attribute :crc32c
+        attribute :etag,                :aliases => "etag"
+        attribute :time_created,        :aliases => "timeCreated"
+        attribute :last_modified,       :aliases => "updated"
+        attribute :generation
+        attribute :metageneration
+        attribute :metadata
+        attribute :self_link,           :aliases => "selfLink"
+        attribute :media_link,          :aliases => "mediaLink"
+        attribute :owner
+        attribute :storage_class, :aliases => "storageClass"
+
+        @valid_predefined_acls = %w(private projectPrivate bucketOwnerFullControl bucketOwnerRead authenticatedRead publicRead)
+
+        def predefined_acl=(new_acl)
+          unless @valid_predefined_acls.include?(new_acl)
+            raise ArgumentError.new("acl must be one of [#{@valid_predefined_acls.join(', ')}]")
+          end
+          @predefined_acl = new_acl
+        end
+
+        # def acl=(new_acl)
+        #   valid_acls = ["private", "projectPrivate", "bucketOwnerFullControl", "bucketOwnerRead", "authenticatedRead", "publicRead"]
+        #   unless valid_acls.include?(new_acl)
+        #     raise ArgumentError.new("acl must be one of [#{valid_acls.join(', ')}]")
+        #   end
+        #   @acl = new_acl
+        # end
+
+        # TODO: Verify
+        def body
+          attributes[:body] ||= last_modified && (file = collection.get(identity)) ? file.body : ""
+        end
+
+        # TODO: Verify
+        def body=(new_body)
+          attributes[:body] = new_body
+        end
+
+        attr_reader :directory
+
+        # TODO: Verify
+        def copy(target_directory_key, target_file_key)
+          requires :directory, :key
+          service.copy_object(directory.key, key, target_directory_key, target_file_key)
+          target_directory = service.directories.new(:key => target_directory_key)
+          target_directory.files.get(target_file_key)
+        end
+
+        # TODO: Verify
+        def destroy
+          requires :directory, :key
+          begin
+            service.delete_object(directory.key, key)
+          rescue Excon::Errors::NotFound
```

Comment:
This is unmodified from the [original storage definition](https://github.com/icco/fog-google/blob/new_storage_api/lib/fog/google/models/storage/file.rb#L47). Should I log the error in both places? Something like:

```ruby
rescue Excon::Errors::NotFound => e
    logger.error e
    logger.error e.backtrace
end
```